nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -50,7 +50,7 @@ in
           if (config.boot.isContainer == false)
           then
             # Trust udev when not in the container
-            [ (subsystemDevice dev) ]
+            optional (dev != null) (subsystemDevice dev)
           else
             # When in the container, check whether the interface is built from other definitions
             if (hasAttr dev cfg.bridges) ||
```

Comment:
@Mic92 if there is a device, it will be rendered as a list with the one device-string, otherwise the empty list is returned.

for the other branches, the empty device needs not be looked at, because `null` will not be part of bridges or bonds or any other aggregated device.